google/cloud/sql/connector/instance_connection_manager.py

```diff
@@ -299,22 +358,22 @@ async def _perform_refresh(self) -> asyncio.Task:
 
         self._current = self._loop.create_task(self._get_instance_data())
         # Ephemeral certificate expires in 1 hour, so we schedule a refresh to happen in 55 minutes.
-        self._next = self._loop.create_task(self._schedule_refresh(_delay))
+        self._next = self._loop.create_task(self._schedule_refresh())
```

Comment:
Can we move the delay calculation into this section? 

I feel like I'm becoming a bit disjointed between how this operates and how the java version operates. This method doesn't seem to actually be performing any refresh, just scheduling it. The schedule refresh doesn't seem to be scheduling, just performing. Is this confusing for you as well?

Reply:
So I've tried moving the delay calculations into this section, but I get this test failure:
```
    async def seconds_until_refresh(self) -> int:
>       expiration = (await self._current).expiration
E       RuntimeError: Task <Task pending coro=<test_InstanceConnectionManager_perform_refresh() running at /Users/shubhadayini/cloud-sql-python-connector/tests/unit/test_instance_connection_manager.py:63> cb=[_run_until_complete_cb() at /Users/shubhadayini/.pyenv/versions/3.7.6/lib/python3.7/asyncio/base_events.py:157]> got Future <Task pending coro=<InstanceConnectionManager._get_instance_data() running at /Users/shubhadayini/cloud-sql-python-connector/google/cloud/sql/connector/instance_connection_manager.py:259>> attached to a different loop
```

It might also be an issue with how the test is written. Calling the delay calculation in schedule refresh makes the error go away.

Reply:
from what I understand schedule_refresh sleeps for `delay` seconds before performing the next refresh. it's not as clean as the executor.submit in Java though